src/Canvas.js

```diff
@@ -168,7 +168,7 @@ var Canvas = React.createClass({
                         || nextProps.columns !== this.props.columns
                         || nextProps.width !== this.props.width
                         || nextProps.cellMetaData !== this.props.cellMetaData
-                        || !shallowEqual(nextProps.style, this.props.style);
+                        || shallowCloneObject(nextProps.style) !== shallowCloneObject(this.props.style);
```

Comment:
Sorry for the drive by commenting (I'm here because we manage a fork we use internally [here](https://github.com/intljusticemission/react-data-grid)), and just noticed this.

This doesn't work right? cloning two objects is _never_ going to result in them being referentially equal, so this will always be `true`.